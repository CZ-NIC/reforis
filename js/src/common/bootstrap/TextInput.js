/*
 * Copyright (C) 2019 CZ.NIC z.s.p.o. (http://www.nic.cz/)
 *
 * This is free software, licensed under the GNU General Public License v3.
 * See /LICENSE for more information.
 */

import React from 'react';
import PropTypes from 'prop-types';

import Input from './Input';


const TextInput = ({...props}) => <Input type="text" {...props}/>;


TextInput.propTypes = {
    /** Field label. */
    label: PropTypes.string.isRequired,
    /** Error text. */
    error: PropTypes.string,
    /** Help text message. */
    helpText: PropTypes.string,
};

export default TextInput;
