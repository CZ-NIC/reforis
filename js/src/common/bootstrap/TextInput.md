Text Bootstrap component input with label and predefined sizes and structure for using in foris forms.

All additional `props` are passed to the `<input type="text">` HTML component.

```js
import {useState} from 'react';
const [value, setValue] = useState('Bla bla');

<TextInput
    value={value}
    label="Some text" 
    helpText="Read the small text!"
    onChange={event => setValue(event.target.value)}
/>
```
