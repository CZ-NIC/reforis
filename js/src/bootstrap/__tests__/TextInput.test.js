/*
 * Copyright (C) 2019 CZ.NIC z.s.p.o. (http://www.nic.cz/)
 *
 * This is free software, licensed under the GNU General Public License v3.
 * See /LICENSE for more information.
 */

import renderer from 'react-test-renderer';
import TextInput from '../TextInput';
import React from 'react';

describe('<TextInput/>', () => {
    it('Render text input', () => {
        const tree = renderer
            .create(
                <TextInput
                    label="Test label"
                    checked
                    helpText="Some help text"
                    value="Some text"
                />
            )
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
});