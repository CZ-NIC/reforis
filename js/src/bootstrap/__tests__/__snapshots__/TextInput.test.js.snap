// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TextInput/> Render text input 1`] = `
<div
  class="form-group row"
>
  <label
    class="form-control-label col-sm-2"
    for="1"
  >
    Test label
  </label>
  <div
    class="col-sm-9"
  >
    <input
      class="form-control "
      id="1"
      type="text"
      value="Some text"
    />
    <div
      class="invalid-feedback"
    />
    <small
      class="form-text text-muted"
    >
      Some help text
    </small>
  </div>
</div>
`;
